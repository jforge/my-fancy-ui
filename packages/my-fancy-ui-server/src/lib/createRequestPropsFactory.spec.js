const mockFs = require('mock-fs');
const createRequestPropsFactory = require('./createRequestPropsFactory.js');

describe('createRequestPropsFactory()', () => {
  beforeEach(() => {
    mockFs(
      {
        '/foo/bar/bundle.js': `
module.exports = {
  getInitialProps: () => ({
    foo: 'bar'
  })
}`
      },
      {
        createCwd: false,
        createTmp: false
      }
    );
  });

  afterEach(() => {
    mockFs.restore();
  });

  it('should return a function.', () => {
    const createRequestProps = createRequestPropsFactory({
      bundlePath: '/foo/bar/bundle.js',
      hypernovaComponentId: 'foo-bar'
    });

    expect(typeof createRequestProps).toBe('function');
  });

  it('should read and evaluate the bundlePath and resolve with a Promise containing the incomming query and initialState.', async () => {
    const createRequestProps = createRequestPropsFactory({
      bundlePath: '/foo/bar/bundle.js',
      hypernovaComponentId: 'foo-bar'
    });
    const query = {baz: 'qux'};

    const hypernovaQuery = await createRequestProps({query});

    expect(hypernovaQuery).toMatchSnapshot();
  });
});
