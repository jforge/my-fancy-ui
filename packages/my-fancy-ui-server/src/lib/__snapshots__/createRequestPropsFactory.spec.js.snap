// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createRequestPropsFactory() should read and evaluate the bundlePath and resolve with a Promise containing the incomming query and initialState. 1`] = `
Object {
  "foo-bar": Object {
    "foo": "bar",
  },
}
`;
